src/main/java/org/jsoup/helper/CSVGenerator.java

```diff
@@ -0,0 +1,89 @@
+import java.util.List;
+
+import org.apache.commons.io.FileUtils;
+import org.jsoup.nodes.HTMLTable;
+import org.jsoup.nodes.TableCell;
+
+/**
+ * Helper class to generate CSV files from a list of HTML tables.
+ * 
+ * This class provides a method to generate CSV files from a list of HTML tables and save them to the specified directory with the given file name prefix.
+ * Each HTML table will be saved as a separate CSV file. The content of each table cell will be written as a separate cell in the CSV file.
+ * Rows are separated by a line break, and columns are separated by a comma.
+ * 
+ * Example Usage:
+ * CSVGenerator csvGenerator = new CSVGenerator();
+ * List<HTMLTable> tables = ...; // populate the list of HTML tables
+ * String filesSavePath = "/csv-files"; // specify the directory to save the CSV files
+ * String filesName = "table"; // specify the file name prefix for the CSV files
+ * csvGenerator.generateCSV(tables, filesSavePath, filesName);
+ * 
+ * @author Noura Alroomi
+ */
+public class CSVGenerator {
+
+    /**
+     * Generates CSV files from a list of HTML tables and saves them to the specified directory with the given file name prefix.
+     * Each HTML table will be saved as a separate CSV file. The content of each table cell will be written as a separate cell in the CSV file.
+     * Rows are separated by a line break, and columns are separated by a comma.
+     * 
+     * @param tables        List of HTMLTable objects representing the tables to write into CSV files
+     * @param filesSavePath Directory path to save the generated CSV files
+     * @param filesName     Prefix for the file names of the CSV files
+     * 
+     * @throws IOException  If an I/O error occurs while creating or writing to the files
+     */
+    public void generateCSV(List<HTMLTable> tables, String filesSavePath, String filesName) throws IOException {
+        // validate the inputs
+        Validate.notEmpty(filesSavePath, "filesSavePath must be set to CSVs save path");
+        Validate.notEmpty(filesName, "filesName must be set to CSVs save prefix file name");
+        Validate.ensureNotNull(tables, "list of HTML tables must be set");
+        Validate.isTrue(tables.size() > 0, "list of HTML tables is empty");
+
+        // create the directory if it doesn't exist
+        File directory = new File(filesSavePath);
+        FileUtils.forceMkdir(directory);
+
+
+        // process each HTML table
+        for (int tableIndex = 0; tableIndex < tables.size(); tableIndex++) {
+            HTMLTable table = tables.get(tableIndex);
+            List<List<TableCell>> tableData = table.getTableData();
+
+            // generate the file name and path for the CSV file
+            String fileName = filesName + "_" + (tableIndex + 1);
+            String filePath = filesSavePath + File.separator + fileName + ".csv";
+
+            // create a FileWriter to write to the CSV file
+            FileWriter writer = new FileWriter(filePath);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *writer*. The resource is referenced at line 82. The resource is closed at line 86. There are other execution paths that do not contain closure statements, for example, when Writer.write throws an exception. Ensure *writer* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).